File: frontend/src/services/analyticsService.ts
import { api } from "./api";

export interface DashboardStats {
  active_goals: number;
  completed_goals: number;
  memories_count: number;
  journal_count: number;
  total_saved: number;
}

export const getDashboard = async (): Promise<DashboardStats> => (await api.get("/analytics/dashboard")).data;

File: backend/app/services/analytics_service.py
from sqlalchemy.orm import Session

from app.models.goal import Goal
from app.models.journal import JournalEntry
from app.models.memory import Memory
from app.services import savings_service


def dashboard_stats(db: Session, couple_id: int, anniversary_date: str | None) -> dict:
    goals = db.query(Goal).filter(Goal.couple_id == couple_id).all()
    memories_count = db.query(Memory).filter(Memory.couple_id == couple_id).count()
    journal_count = db.query(JournalEntry).filter(JournalEntry.couple_id == couple_id).count()
    money = savings_service.summary(db, couple_id)

    return {
        "active_goals": len([g for g in goals if g.status == "active"]),
        "completed_goals": len([g for g in goals if g.status == "completed"]),
        "memories_count": memories_count,
        "journal_count": journal_count,
        "total_saved": money["balance"],
    }
